File: life-assisitant-ui-desktop/src/api/timeline.ts
import request from './request'
import type { TimelineQuery, TimelineResp } from './types'

/**
 * 生活时间线 API
 * 规范：md/design-260917/04-功能深化-私人管家.md · §4
 * - 聚合任务完成 / 习惯打卡 / 记账（不含转账）/ 心情，按完整时间戳倒序
 * - GET /timeline?date=YYYY-MM-DD&page=&page_size=
 */
export const timelineApi = {
  /** 获取某天时间线（跨模块聚合，倒序） */
  getTimeline(params?: TimelineQuery) {
    return request.get<unknown, TimelineResp>('/timeline', { params })
  },
}

File: life-assisitant-api/internal/controller/timeline.go
// Package controller 生活时间线控制器
package controller

import (
	"github.com/gogf/gf/v2/net/ghttp"

	"github.com/life-assistant/api/internal/consts/ecode"
	"github.com/life-assistant/api/internal/model/dto"
	"github.com/life-assistant/api/internal/response"
	"github.com/life-assistant/api/internal/service"
)

// TimelineController 时间线控制器
type TimelineController struct{}

var Timeline = &TimelineController{}

// Fetch GET /api/v1/timeline?date=YYYY-MM-DD&page=&page_size=
func (c *TimelineController) Fetch(r *ghttp.Request) {
	var req dto.TimelineReq
	if err := r.Parse(&req); err != nil {
		response.Error(r, ecode.ValidationFailed, err)
		return
	}
	out, err := service.Timeline().Fetch(r.Context(), &req)
	if err != nil {
		writeError(r, err)
		return
	}
	response.Success(r, out)
}
